context/auth.tsx

```diff
@@ -1,16 +1,42 @@
 import React from "react"
-import { loadFirebase } from "./firebase"
+import { loadFirebase, userType } from "./firebase"
 import axios from "../util/axios"
+import { ProfileSerializer } from "../types/backend"
+import { AxiosResponse } from "axios"
 
-const AuthContext = React.createContext({})
+export type tokenType = string | null
+export type profileType = ProfileSerializer | null
+export type loadingType = boolean
 
-export const AuthProvider = ({ children }) => {
-    const [firebaseUser, setFirebaseUser] = React.useState(null)
-    const [token, setToken] = React.useState(null)
-    const [profile, setProfile] = React.useState(null)
-    const [loading, setLoading] = React.useState(true)
+const AuthContext = React.createContext<{
+    firebaseUser: userType
+    setFirebaseUser: (user: userType) => void
+    token: tokenType
+    setToken: (token: tokenType) => void
+    profile: profileType
+    setProfile: (profile: profileType) => void
+    loading: loadingType
+    handleSignIn: () => void
+    handleLogout: () => void
```

Comment:
If possible, try to order these in some order. Like variables, void functions, functions with arguments. Just for better visibility.

Reply:
done